### PR TITLE
Support external templates in amp-date-countdown

### LINKED ISSUE #18360: amp-date-countdown: Support 'template' attribute
amp-date-countdown inside an amp-list generates this alert: The tag 'template' may not appear as a descendant of tag 'template'. 

The fact is: in this case it works fine. I can fill end-date attribute inside a amp-counterdown tag with a custom end date of a READER_ID and amp-date-countdown tag generates the correct counterdown. 

Two ways to solve this question:
1. Allow tag template as descendat of tag template when amp-counterdown is descendat of amp-list; or
2. Implement a src attribute inside a counterdown to output custom counterdown to different READER_ID users.

To reproduce this issue: 
1. Create an amp-counterdown tag inside template tag of amp-list.
2. Set amp-list src attribute to return an end-date like {"items":[{"end-date":"2018-09-25T18:34:08.000Z"}]};
3. Put {{end-date}} inside end-date attribute in a descendant amp-counterdown tag
4. Set time display inside template tag of amp-counterdown tag like {{hh}}:{{mm}}:{{ss}} 
5. It will works fine but the validator will display validation error message;

AMP Version 1809142227020
macOS High Sierra - Chrome Version 69.0.3497.100

### PR DESCRIPTION
**Pull request by choumx**
Fixes #18360. Tested on examples/amp-date-countdown.amp.html.

### CODE REVIEW

File: extensions/amp-date-countdown/validator-amp-date-countdown.protoascii [hunk 1]
```diff
@@ -64,6 +64,9 @@ tags: {  # <amp-date-countdown>
     name: "offset-seconds"
     value_regex: "-?\\d+"
   }
+  # TODO(gregable): Implement validation that requires the template attr value
+  # to reference the id of an existing template element.
+  attrs: { name: "template" }
```
Comment (choumx) [inline extensions/amp-date-countdown/validator-amp-date-countdown.protoascii]: @honeybadgerdontcare Validator change FYI.

Comment (honeybadgerdontcare) [reply extensions/amp-date-countdown/validator-amp-date-countdown.protoascii]: @choumx I'm not following the todo. Are you allowing an attribute that needs additional validation in the validator engine for it to work properly? Usually we wait to add that engine change before allowing the attribute. Or do we not care if that is enforced in validation and if it does get enforced later it is okay to invalidate pages that do it wrong?

Comment (choumx) [reply extensions/amp-date-countdown/validator-amp-date-countdown.protoascii]: I copied this TODO from another extension that has the same feature: https://github.com/ampproject/amphtml/blob/617b7e4e25aa578756c5af941c2748c50abd8004/extensions/amp-list/validator-amp-list.protoascii#L74-L76

Comment (choumx) [reply extensions/amp-date-countdown/validator-amp-date-countdown.protoascii]: This feature doesn't require additional validator changes to work properly. 

The comment describes a nice-to-have additional validation step. I imagine that we'd either make it a validator warning or error, depending on the # of pages affected.

Comment (honeybadgerdontcare) [reply extensions/amp-date-countdown/validator-amp-date-countdown.protoascii]: Thank you for the explanation. Filed b/116870484 for tracking.

**Comment by leonalicious** [comment]
thank you for the update, appreciate it

**Pull request merged**